lib/clone_kit/cloners/mongoid_ruleset_cloner.rb

```diff
@@ -107,9 +107,34 @@ def save_or_fail(attributes)
         if model_that_we_wont_save.valid?
           model_klass.collection.insert(attributes)
         else
-          details = model_that_we_wont_save.errors.full_messages.to_sentence
-          id = attributes["_id"]
-          current_operation.error("#{model_klass} #{id} failed model validation and was not cloned: #{details}")
+          report_errors(attributes, model_that_we_wont_save)
+        end
+      end
+
+      def report_errors(attributes, model)
+        model_error = model.errors.full_messages.to_sentence
+        embedded_errors = collect_embedded_errors(model)
+        id = attributes["_id"]
+
+        current_operation.error("#{model_klass} #{id} failed model validation and was not cloned: #{model_error}")
+
+        embedded_errors.map do |e|
+          current_operation.error("[#{model_klass} #{id}]: #{e}")
+        end
+      end
+
+      def collect_embedded_errors(model)
+        embedded_fields = model.associations.select { |_key, assoc| assoc.embedded? }.keys
+        embedded_documents = embedded_fields.flat_map { |field| model.send(field) }
```

Comment:
`..` => `.`

Is there a need to flat_map here?  Should this just be an array of keys already?

Reply:
You're right. I think I want to call `flatten` on the result as `model.send(field)` could return an array